### PR TITLE
Compositional: Rewrite flash to use vapor fraction, add tests for Rachford-Rice

### PR DESCRIPTION
**Pull request by moyner**
Draft PR for discussion purposes.

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this opm-models=857 please

**Comment by bska** [comment]
Just a quick question: Is this intended as a replacement for or a supplement to the existing PR #3745, or is there perhaps no relation between the two PRs at all?

**Comment by GitPaean** [comment]
> Just a quick question: Is this intended as a replacement for or a supplement to the existing PR https://github.com/OPM/opm-common/pull/3745, or is there perhaps no relation between the two PRs at all?

They are partly related/overlapped.  Some of the PR in https://github.com/OPM/opm-common/pull/3745 might still go in.  But this PR should be reviewed and merged first. Then https://github.com/OPM/opm-common/pull/3745 will be updated based on the new master and new testing.

**Comment by bska** [comment]
> > Just a quick question: Is this intended as a replacement for or a supplement to the existing PR #3745, or is there perhaps no relation between the two PRs at all?
> 
> They are partly related/overlapped. Some of the PR in #3745 might still go in. But this PR should be reviewed and merged first. Then #3745 will be updated based on the new master and new testing.

Understood.  Thanks for the quick response.

**Comment by GitPaean** [comment]
Furthermore,  with some discussion, ssi + newton might not be essential or the main solver. So PR https://github.com/OPM/opm-common/pull/3745 might not be the things necessary for other development. It is something good to have or for future.  But we need to review and test this PR first and check afterwards.

File: opm/material/constraintsolvers/PTFlash.hpp [hunk 1]
```diff
@@ -357,12 +272,12 @@ class PTFlash
 
             // Print iteration info
             if (verbosity == 3 || verbosity == 4) {
-                std::cout << std::setw(10) << iteration << std::setw(16) << Opm::abs(delta) << std::setw(16) << L << std::endl;
+                std::cout << std::setw(10) << iteration << std::setw(16) << Opm::abs(delta) << std::setw(16) << V << std::endl;
             }
             // Check for convergence
-            if ( Opm::abs(delta) < 1e-10 ) {
+            if ( Opm::abs(r) < tol || iteration == itmax) {
```
Comment (GitPaean) [inline opm/material/constraintsolvers/PTFlash.hpp:278]: `iteration == itmax` will never be `true`.

Comment (GitPaean) [reply opm/material/constraintsolvers/PTFlash.hpp:278]: it is fixed in https://github.com/moyner/opm-common/pull/1 .

**Comment by GitPaean** [comment]
closing this one in favor of #3956

**Pull request closed without merging**